src/python/bot/minimizer/js_minimizer.py

```diff
@@ -30,47 +37,79 @@ def _execute(self, data):
 
     brace_stack = []
     paren_stack = []
-    last_start_of_line_token = 0
 
     for index, token in enumerate(testcase.tokens):
       if token == '{':
-        brace_stack.append((last_start_of_line_token, index))
+        brace_stack.append(index)
 
       elif token == '}' and brace_stack:
-        previous_line_start, previous_end = brace_stack.pop()
-
-        # Move |previous_line_start| to the first non-empty string.
-        while previous_line_start > 0:
-          if testcase.tokens[previous_line_start].strip():
-            break
-          previous_line_start -= 1
 
         # Two hypotheses for tokens grouped by curly braces:
-        # 1) Remove from start of line to curly brace and the closing brace.
+        # 1) Remove from start of line to open brace and the closing brace.
         #    e.g.: if (statement_that_evaluates_to_true) { crash() } -> crash()
-        hypothesis = list(range(previous_line_start,
-                                previous_end + 1)) + [index]
+        open_brace_index = brace_stack.pop()
+
+        # Find the first non-empty token prior to the starting brackets.
+        token_before_bracket = (
```

Comment:
Nit: step_back_while on this line to avoid the outermost pair of parens. Same for other calls.